ngraph/core/src/op/psroi_pooling.cpp

```diff
@@ -54,29 +54,69 @@ bool ngraph::op::v0::PSROIPooling::visit_attributes(AttributeVisitor& visitor)
 
 void op::PSROIPooling::validate_and_infer_types()
 {
-    auto input_et = get_input_element_type(0);
+    auto feat_maps_et = get_input_element_type(0);
+    auto coords_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(this,
+                          feat_maps_et.is_real(),
+                          "Feature maps' data type must be floating point. Got " +
+                              feat_maps_et.get_type_name());
+    NODE_VALIDATION_CHECK(this,
+                          coords_et.is_real(),
+                          "Coords' data type must be floating point. Got " +
+                              coords_et.get_type_name());
+    NODE_VALIDATION_CHECK(this,
+                          m_mode == "average" || m_mode == "bilinear",
+                          "Expected 'average' or 'bilinear' mode. Got " + m_mode);
+    NODE_VALIDATION_CHECK(this, m_group_size > 0, "group_size has to be greater than 0");
```

Comment:
Please, make the shape inference function as much specific as possible. It is possible to calculate some of the output tensor dimensions if one of the input dimensions is dynamic. For example, if the channel dimension of the `feat_maps_et` tensor is dynamic we still can calculate height and width of the output tensor as they are always equal to `m_group_size` .  Also, add unit tests to cover various scenarios when different dimensions are dynamic.